src/test/java/com/teragrep/pth10/TokenizerTest.java

```diff
@@ -142,4 +156,52 @@ public void tokenize3() {
                         }
                 );
     }
+
+    @Test
+    @DisabledIfSystemProperty(
+            named = "skipSparkTest",
+            matches = "true"
+    )
+    public void testRegexTokenizerNotSupportedFormat() {
+        Properties properties = new Properties();
+        properties.put("dpl.pth_06.bloom.pattern", "testRegex");
+        Config config = ConfigFactory.parseProperties(properties);
+
+        TokenizerStep step = new TokenizerStep(config, AbstractTokenizerStep.TokenizerFormat.STRING, "_raw", "result");
+        SparkSession spark = streamingTestUtil.getCtx().getSparkSession();
+        Assertions
+                .assertThrows(UnsupportedOperationException.class, () -> step.get(spark.read().schema(testSchema).csv(spark.emptyDataset(Encoders.STRING()))));
+    }
+
+    @Test
+    @DisabledIfSystemProperty(
+            named = "skipSparkTest",
+            matches = "true"
+    )
+    public void testRegexTokenizerSelection() {
+        String regex = "^\\d+";
+        Pattern pattern = Pattern.compile(regex);
+        Properties properties = new Properties();
+        properties.put("dpl.pth_06.bloom.pattern", regex);
+        Config config = ConfigFactory.parseProperties(properties);
+
+        TokenizerStep step = new TokenizerStep(
+                config,
+                AbstractTokenizerStep.TokenizerFormat.BYTES,
+                "source",
+                "bytetokens"
+        );
+        SparkSession spark = streamingTestUtil.getCtx().getSparkSession();
+        Dataset<Row> result = step.get(spark.read().schema(testSchema).json(regexTestFile));
+        // first row value: 127.0.0.0
+        List<Object> list = result.select("bytetokens").first().getList(0);
+        List<String> tokens = new ArrayList<>();
+        for (Object o : list) {
+            tokens.add(new String((byte[]) o, StandardCharsets.UTF_8));
+        }
+        Assertions.assertEquals(4, list.size());
```

Comment:
While the expected result's size=4, with values 127, 0, 0 and 0, maybe remove duplicates from list to make sure there is only 127 and 0, because here the other two values could be anything else

Reply:
switched list to set with expect result size 2 and values 127, 0